walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+
+
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+      Collection<String> attendees = request.getAttendees();
+      Collection<String> optional = request.getOptionalAttendees();
+      long duration = request.getDuration();
+      //EDGE CASES
+      //return whole day if the meeting to schedule requires no attendees
```

Comment:
That's sounds illogically :) If there is no attendees - no meetings should be scheduled?

Reply:
+1

Reply:
I wrote this edge case based on the test provided for us
```
@Test
  public void optionsForNoAttendees() {
    MeetingRequest request = new MeetingRequest(NO_ATTENDEES, DURATION_1_HOUR);

    Collection<TimeRange> actual = query.query(NO_EVENTS, request);
    Collection<TimeRange> expected = Arrays.asList(TimeRange.WHOLE_DAY);

    Assert.assertEquals(expected, actual);
  }
```
did I misunderstand the desired functionality?

Reply:
done